shm/src/Cargo.cpp

```diff
@@ -9,16 +11,20 @@ Cargo::Cargo(const std::string& name, size_t amount, size_t basePrice)
 {}
 
 Cargo& Cargo::operator+=(const size_t amount) {
+    if (amount_ + amount > std::numeric_limits<size_t>::max()) {
```

Comment:
A huge number of cargo! do we need so many? The idea was to return ::max() to inform about an error case - not to compare.